src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/oasys/controller/OasysAssessmentController.kt

```diff
@@ -194,7 +251,17 @@ class OasysAssessmentController(
     @PathVariable
     oasysAssessmentPK: String,
   ): OasysAssessmentResponse {
-    val lockedVersion = oasysAssessmentService.lock(oasysAssessmentPK)
-    return OasysAssessmentResponse.from(lockedVersion.assessment.uuid, lockedVersion.versionNumber)
+    return oasysAssessmentService.find(oasysAssessmentPK)
+      .let {
+        assessmentVersionService.find(
+          AssessmentVersionCriteria(it.assessment.uuid, Tag.validatedTags()),
+        )
+      }
+      .let {
+        assessmentVersionService.lock(it)
```

Comment:
Using `run` *could* be nice here as we call a single param function inside the `let`, for example;

```kotlin
.run(assessmentVersionService::lock)
```

Reply:
good idea! I've applied it to a few other places as well